File: Top Interview Questions/Array/Plus One.go
// https://leetcode.com/explore/interview/card/top-interview-questions-easy/92/array/559/

func plusOne(digits []int) []int {

    for index := len(digits) - 1; index >= 0; index-- {
        value := digits[index]

        if index == 0 && value == 9 {
            digits[index] = 1
            digits = append(digits, 0)
            break
        }

        if value == 9 {
            digits[index] = 0
        } else {
            digits[index] = value + 1
            break
        }
    }

    return digits

}

File: Array/Plus One.swift
// https://leetcode.com/explore/interview/card/top-interview-questions-easy/92/array/559/

class Solution {
    func plusOne(_ digits: [Int]) -> [Int] {
        var result = digits
        var i = digits.count - 1
        while (i >= 0) {
            if result[i] == 9 {
                result[i] = 0
                if i == 0 {
                    result.insert(1, at: 0)
                    return result
                }
                i -= 1
            } else {
                result[i] += 1
                return result
            }
        }
        return result
    }
}
